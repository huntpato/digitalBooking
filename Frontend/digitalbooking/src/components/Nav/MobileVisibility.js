import { createContext } from "react";

export const MobileVisibilty  = createContext({
    
});