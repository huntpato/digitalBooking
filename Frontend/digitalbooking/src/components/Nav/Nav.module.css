nav {
  align-items: center;
  display: flex;
  flex-direction: row;
  height: 80px;
  justify-content: space-between;
  padding: 20px 40px 10px 40px;
  position: fixed;
  transition: 0.5s;
  width: 100%;
  z-index: 2;
  -webkit-animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
  animation: fade-in 0.3s cubic-bezier(0.39, 0.575, 0.565, 1) both;
}

.color1 {
  background-color: var(--background-white);
}

.color2 {
  background-color: var(--background-white);
  height: 80px;
  transition: 0.5s;
}

.hide {
  display: none;
}

.signedUp {
  text-decoration: none;
}

nav button {
  background-color: var(--background-white);
  border-radius: 5px;
  border: 1.4px solid var(--principal-color);
  color: var(--principal-color);
  cursor:pointer;
  font-weight: 700;
  margin-left: 50px;
  padding: 6px 30px 5px 30px;
}

nav img {
  height: 45px;
  margin-left: 35px;
  cursor:pointer;
}

nav input {
  background-color: var(--background-color);
  border-radius: 5px;
  border: 0px solid var(--principal-color);
  font-family: "Roboto", "FontAwesome", sans-serif;
  list-style: none;
}

nav i {
  background-color: var(--principal-color);
  border-radius: 20px;
  color: var(--background-white);
  font-size: 10px;
  padding: 8px;
}

.searchNav {
  background-color: var(--background-white);
  border-radius: 20px;
  border: 0.5px solid var(--background-color);
  box-shadow: 0px 0px 0px 1px rgba(234, 234, 234, 0.75);
  padding: 6px 5px 6px 10px;
}

.searchInput {
  display: none;
}

.scrolled {
  display: block;
  padding-left: 40px;
}

.signedUptrue {
  display: none;
}

.signedOut {
  display: block;
}

.hamburguesita {
  display:none;
}

.visible{
  display: block;
}

@media (max-width: 768px) {
  nav img {
    height: 35px;
    margin-left: 4%;
  }

  .searchNav {
    width: 180px;
  }
}

@media (max-width: 705px) {
  
  .searchNav {
    display:none;
  }

  .scrolled{
    display:none
  }
}

@media (max-width: 480px) {

  nav img {
    height: 30px;
    margin-left: 0%;
  }

  .hamburguesita{
    display: block;
    height: 23px;
    width: 23px;
  }
}


@-webkit-keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
@keyframes fade-in {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
