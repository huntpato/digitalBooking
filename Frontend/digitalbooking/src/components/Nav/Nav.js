import React, {
  Fragment,
  useContext,
  useState,
  useMemo,
  useEffect,
} from "react";
import logo from "../../assets/logo.png";
import menu from "../../assets/hamburguer.svg";
import { MobileVisibilty } from "./MobileVisibility";
import { SignedUpNav } from "../SignedUpNav/SignedUpNav";
import { NavButtons } from "../NavButtons/NavButtons";
import { MenuMobile } from "../MenuMobile/MenuMobile";
import { useHistory } from "react-router-dom";
import { UserContext } from "../UserContext/UserContext";
import { NavContext } from "./NavContext";
import styles from "./Nav.module.css";


export const Nav = () => {
  const { user } = useContext(UserContext);
  const [search, setsearch] = useState(true);
  const {signedbar, setSignedbar}= useContext(NavContext);
  const [visibility, setVisibility] = useState(false);
  const value = useMemo(
    () => ({ visibility, setVisibility }),
    [visibility, setVisibility]
  );
  const { searchNav, searchInput, scrolled, color1, color2, hamburguesita } =
    styles;
  let history = useHistory();

  const changeVisibility = () => {
    if (window.scrollY >= 80) {
      setsearch(false);
    } else {
      setsearch(true);
    }
  };

  const handleClick = (e) => {
    e.preventDefault();
    history.push("/");
  };

  const handleMenu = (e) => {
    e.preventDefault();
    setVisibility(true);
  };

  window.addEventListener("scroll", changeVisibility);

  return (
    <Fragment>
      <MobileVisibilty.Provider value={value}>
        <MenuMobile />
      </MobileVisibilty.Provider>
      <nav className={search ? color1 : color2}>
        <img onClick={handleClick} src={logo} alt="Logo" />
        <section className={search ? searchInput : scrolled}>
          <i className="fas fa-search"></i>
          <input
            type="text"
            className={searchNav}
            placeholder="Encontrá tu lugar"
          >
          </input>
        </section>
          { user !=null ? <SignedUpNav /> : <NavButtons /> }
        <img
          className={hamburguesita}
          alt="ícono menú"
          src={menu}
          onClick={handleMenu}
        />
      </nav>
    </Fragment>
  );
};
