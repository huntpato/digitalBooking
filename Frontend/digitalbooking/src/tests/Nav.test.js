import React from 'react';
import { shallow } from 'enzyme';
import { Nav } from '../components/Nav/Nav'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

jest.mock('react-router-dom', () => ({
    useParams: jest.fn().mockReturnValue({ nifUuid: 'nif123' }),
    useHistory: jest.fn(),
    useLocation:jest.fn().mockReturnValue({
      pathname: "/register",
      search: "",
      hash: "",
      state: null,
      key: "5nvxpbdafa",
    }),
  }));

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush,
  }),
}));

describe('Pruebas en componente  Nav />', ()=> {
    const text = "Ver mas";
    const wrapper= shallow( <Nav text='Ver mas'/>);

    test('Comprobar que  Nav /> renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que el componente carga la clase de CSS correctamente', () => {

        const nav = wrapper.find('nav');
        expect(nav.hasClass('color1')).toBe(true);
    })
    
    test('Verificar que el logo se renderiza correctamente', () => {
        const logo= "logo.png";
        expect (wrapper.find("img").at(0).prop("src")).toEqual(logo);
    })
    
    test('Verificar que aparece el componente de búsqueda al hacer scroll', () => {
      const spyScrollTo = jest.fn();
      Object.defineProperty(global.window, 'scrollTo', { value: spyScrollTo });
      Object.defineProperty(global.window, 'scrollY', { value: 400 });
       
        expect (wrapper.find("input").props().placeholder).toContain("Encontrá tu lugar");
    })

    test('Simular click menu', () => {
        
    const setVisibility = jest.fn();   
    wrapper.find(".hamburguesita").simulate("click", {
        preventDefault: () => {
        }
    });
    expect(setVisibility).not.toHaveBeenCalled()

        
    })
    
    

})
