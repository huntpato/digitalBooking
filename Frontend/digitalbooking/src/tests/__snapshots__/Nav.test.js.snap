// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente  Nav /> Comprobar que  Nav /> renderiza correctamente. 1`] = `
<Fragment>
  <ContextProvider
    value={
      Object {
        "setVisibility": [Function],
        "visibility": false,
      }
    }
  >
    <MenuMobile />
  </ContextProvider>
  <nav
    className="color1"
  >
    <img
      alt="Logo"
      onClick={[Function]}
      src="logo.png"
    />
    <section
      className="searchInput"
    >
      <i
        className="fas fa-search"
      />
      <input
        className="searchNav"
        placeholder="Encontrá tu lugar"
        type="text"
      />
    </section>
    <NavButtons />
    <img
      alt="ícono menú"
      className="hamburguesita"
      onClick={[Function]}
      src="hamburguer.svg"
    />
  </nav>
</Fragment>
`;
